addon/mixins/ajax-request.js

```diff
@@ -520,15 +520,19 @@ export default Mixin.create({
    * @return {Object | AjaxError} response
    */
   handleResponse(status, headers, payload, requestData) {
-    let error;
-
     if (this.isSuccess(status, headers, payload)) {
       return payload;
     }
 
     // Allow overriding of error payload
     payload = this.normalizeErrorResponse(status, headers, payload);
 
+    return this._createCorrectError(status, headers, payload, requestData);
+  },
+
+  _createCorrectError(status, headers, payload, requestData) {
```

Comment:
What's the reason for breaking this out into a separate method, rather than just adding the logic to the existing `handleResponse`?

Reply:
I actually came from a request you had: https://github.com/ember-cli/ember-ajax/pull/275#pullrequestreview-32876092 
Or did I misunderstand something there?

Reply:
That's fair 😅 That review was so long ago. I'm fine with merging as-is. I'm working on a pretty major refactor so might clean it up later if it's bugging me.